src/ser/mod.rs

```diff
@@ -96,20 +95,22 @@ impl<'a> Serializer<'a> {
 // which take 200+ bytes of ROM / Flash
 macro_rules! serialize_unsigned {
     ($self:ident, $N:expr, $v:expr) => {{
-        let mut buf: [u8; $N] = unsafe { super::uninitialized() };
-
-        let mut v = $v;
         let mut i = $N - 1;
-        loop {
-            buf[i] = (v % 10) as u8 + b'0';
-            v /= 10;
-
-            if v == 0 {
-                break;
-            } else {
-                i -= 1;
+        let mut v = $v;
+        let buf = {
+            let mut buf: [MaybeUninit<u8>; $N] = unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
Isn't this is undefined behavior? You cannot assume initialization on uninitialized memory before initialization.

Reply:
`MaybeUninit` type doesn't require any initialization, reference implementation here: https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element

Reply:
Ah, you're correct because this is an array of MaybeUninit as opposed to a `MaybeUninit<[u8; N]>` - in any case a [`heapless::Vec`](https://docs.rs/heapless/0.7.3/heapless/struct.Vec.html) will make this much simpler.